### PR TITLE
Removing tech-preview header and updating documentation for retrievers and RRF

### PR DESCRIPTION
**Pull request by pmpailis**
Now that the rework of RRF and retrievers in general has been merged, it's probably time to remove the tech-preview header from the documentation in time for `8.16` and mark the feature as GA.

### CODE REVIEW

File: docs/reference/search/search-your-data/retrievers-overview.asciidoc [hunk 1]
```diff
@@ -1,9 +1,7 @@
 [[retrievers-overview]]
 === Retrievers
 
-preview::[]
-
-A retriever is an abstraction that was added to the Search API in *8.14.0*.
+A retriever is an abstraction that was added to the Search API in *8.14.0* and has been made GA in *8.16.0*.
```
Comment (leemthompo) [inline docs/reference/search/search-your-data/retrievers-overview.asciidoc:4]:
```suggestion
A retriever is an abstraction that was added to the Search API in *8.14.0* and was made GA in *8.16.0*.
```

Comment (leemthompo) [reply docs/reference/search/search-your-data/retrievers-overview.asciidoc:4]: very nitty nit :)

**Review by leemthompo — APPROVED** [review]
LGTM docs-wise

File: docs/reference/search/rrf.asciidoc [hunk 2]
```diff
@@ -1,8 +1,6 @@
 [[rrf]]
 === Reciprocal rank fusion
 
-preview::["This functionality is in technical preview and may be changed or removed in a future release. The syntax will likely change before GA. Elastic will work to fix any issues, but features in technical preview are not subject to the support SLA of official GA features."]
```
Comment (benwtrent) [inline docs/reference/search/rrf.asciidoc:4]: We just need to make sure that we CLEARLY communicate that `rank: rrf` is still tech preview and will be removed. I am not sure what things link to this doc page, but we need to be absolutely clear about that.

**Review by benwtrent — COMMENTED** [review]
I think this is good, but we need to be crystal clear about sub-searches & rrf rank. Those need to stay flagged as tech preview & we need to indicate that they will be removed.

File: docs/reference/search/rrf.asciidoc (hunk 2 above)
Comment (pmpailis) [reply docs/reference/search/rrf.asciidoc:4]: All references to docs have been removed when we introduced the rrf retriever [here](https://github.com/elastic/elasticsearch/pull/105470#discussion_r1517535806). Should we add a warning header that `rank: rrf` is a now-deprecated option? 
I've also opened [this PR](https://github.com/elastic/elasticsearch/pull/114854) to add deprecation headers when using `rank: rrf` and/or `sub_searches`

File: docs/reference/search/search-your-data/retrievers-overview.asciidoc [hunk 3]
```diff
@@ -1,9 +1,7 @@
 [[retrievers-overview]]
 === Retrievers
 
-preview::[]
-
-A retriever is an abstraction that was added to the Search API in *8.14.0*.
+A retriever is an abstraction that was added to the Search API in *8.14.0* and was made GA in *8.16.0*.
```
Comment (benwtrent) [inline docs/reference/search/search-your-data/retrievers-overview.asciidoc:4]:
```suggestion
A retriever is an abstraction that was added to the Search API in *8.14.0* and was made generally available in *8.16.0*.
```

**Review by benwtrent — APPROVED** [review]

**Comment by pmpailis** [comment]
@elasticmachine update branch

**Pull request merged**